### PR TITLE
feat: Add Option2 exercise

### PR DESCRIPTION
**Pull request by sanjaykdragon**
I added another exercise for Option<T>, as promised.
Let me know if there are any problems with this.
This one is specifically about if let / while let

### CODE REVIEW

**Comment by sanjaykdragon** [comment]
@fmoko

**Comment by sanjaykdragon** [comment]
> 
> 
> Hmm, I think this exercise could be structured in a different way. As I see it, you just need to wrap the variables in `Some` to pass right now, whereas I think a better approach would be to let the user write their own `if let` and `while let` statements. Also, you should add a couple of comments about which lines the user should try to edit to nudge them in the right direction.

Alright, this is a better idea. I was thinking of doing that, but I felt that leaving the while let / if let sections blank would result in people just changing the entire code. Will change the style now

File: exercises/option/option2.rs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// option2.rs
+// Make me compile! Execute `rustlings hint option2` for hints
+
+// I AM NOT DONE
+
+fn main() {
+    let optional_value = Some(String::from("rustlings"));
+    //make this an if let statement - value is "Some" type
```
Comment (fmoko) [inline exercises/option/option2.rs:8]:
```suggestion
    // Make this an if let statement whose value is "Some" type
```

File: exercises/option/option2.rs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    value = optional_value {
+        println!("the value of optional value is: {}", value);
```
Comment (fmoko) [inline exercises/option/option2.rs:10]:
```suggestion
        println!("The optional value contains: {}", value);
```

File: exercises/option/option2.rs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    } else {
+        println!("optional value does not have a value!");
```
Comment (fmoko) [inline exercises/option/option2.rs:12]:
```suggestion
        println!("The optional value doesn't contain anything!");
```

File: exercises/option/option2.rs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    }
+
+    let mut optional_values_vec: Vec<Option<i8>> = Vec::new();
+    for x in 1..10 {
+        optional_values_vec.push(Some(x));
+    }
+
+    // make this a while let statement - remember that vector.pop also adds another layer of Option<T>
```
Comment (fmoko) [inline exercises/option/option2.rs:20]:
```suggestion
    // Make this a while let statement - remember that `pop` also adds another layer of `Option<T>`
```

File: exercises/option/option2.rs [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    // you can stack Option<T>'s into while let and if let
```
Comment (fmoko) [inline exercises/option/option2.rs:21]:
```suggestion
    // You can stack `Option<T>`'s into while let and if let
```

**Comment by sanjaykdragon** [comment]
added your nits

**Comment by fmoko** [comment]
Thanks!

**Pull request merged**